overlord/configstate/config/helpers.go

```diff
@@ -35,14 +35,23 @@ import (
 
 var validKey = regexp.MustCompile("^(?:[a-z0-9]+-?)*[a-z](?:-?[a-z0-9])*$")
 
+// ValidateKey checks if the provided key matches the validKey regex pattern.
+// It returns an error if the key is invalid, otherwise nil.
+func ValidateKey(key string) error {
+	if !validKey.MatchString(key) {
+		return fmt.Errorf("invalid option name: %q", key)
```

Comment:
Should this say "option"? I think realistic errors will now say something like:
```
error: invalid configuration key: invalid option name: blah
```

I think we can drop one of the two `fmt.Errorf`'s (perhaps the outer one).

Reply:
Agree! Since we're just validating the key, I'm going to change this message to `invalid configuration key:` and drop the other error message.